src/main/java/org/montclairrobotics/sprocket/SprocketRobot.java

```diff
@@ -68,11 +69,14 @@ public final void disabledInit() {
 
     @Override
     public final void autonomousInit() {
-    	selectedAutoMode = (AutoMode) chooser.getSelected();
+	    userAutonomousSetup();
+    	selectedAutoMode =  autoModes[(Integer) chooser.getSelected()];
```

Comment:
chooser.getSelected returns the currently selected object. Why are you doing it this way.

Reply:
No this is correct nevermind